app/js/components/queryBuilder/queryBuilder.component.js

```diff
@@ -225,33 +225,32 @@ const QueryBuilderComponent = {
 
                       if(vm[`${vm.currentState()}Set`].length > 3) vm.progressBar.advance(false);
                       _showIndicatorButton();
-
-                      return vm[`${vm.currentState()}Set`];
                     }
+                  });
 
-                    return false;
-                };
+                  return vm[`${vm.currentState()}Set`];
+                }
```

Comment:
Since this is a function expression instead of function declaration, it is probably better to have `;` at the end (although in general `;` is optional in JS).